core/src/main/java/org/apache/calcite/sql/fun/SqlTrimFunction.java

```diff
@@ -154,6 +155,25 @@ public SqlCall createCall(
     return super.createCall(functionQualifier, pos, operands);
   }
 
+  /**
+   * Returns either a padding char literal or an empty byte literal depending
+   * on the type of {@code node} passed in.
+   *
+   * @param node The node whose type is checked
+   * @param pos The parser position
+   *
+   * @return The null literal
+   */
+  private static SqlLiteral getNullLiteral(SqlNode node, SqlParserPos pos) {
+    if (node instanceof SqlLiteral) {
```

Comment:
nit: Reduce nesting by reversing polarity and returning `createCharString` early.